### ISSUE TITLE
Websocket Handling should be evented, not polled - to avoid 200ms latency

### ISSUE THREAD

**Issue by cefn** [eclipse/mosquitto#147]
The 100ms delay in handling events, coded into...
https://github.com/eclipse/mosquitto/blob/d20355c8ac64746876bdfd0c47f9ac6096bee9dc/src/loop.c
...as...
```
#ifndef WIN32
		sigprocmask(SIG_SETMASK, &sigblock, &origsig);
		fdcount = poll(pollfds, pollfd_index, 100);
		sigprocmask(SIG_SETMASK, &origsig, NULL);
#else
		fdcount = WSAPoll(pollfds, pollfd_index, 100);
#endif
```

...means that 'round-trip' latency  is extremely low, systematically taking 200ms for a sent message to be received, as described at this Eclipse bug...

https://bugs.eclipse.org/bugs/show_bug.cgi?id=482714

There is a workaround by bringing down the poll frequency from 100 to 4 or 2 milliseconds, but this starts to use up the majority of the host's CPU, so a polling strategy should ideally be avoided in order to improve responsiveness without cost of the CPU. This means replacing the poll with some form of wait behaviour, which unblocks the thread on receiving an event to handle.

**Comment by karlp** [comment]
but poll is an event call?  the 100 there is just when to exit if no events were received

**Comment by ralight** [comment]
This is a duplicate of https://github.com/eclipse/mosquitto/issues/23

@karlp you're correct in principle, but the websockets sockets aren't part of the pollfds set and so don't trigger the poll call. It would be very intrusive to fix in the current code. It'll all be sorted out with a proper event loop library.

**Comment by cefn** [comment]
I must have been searching for the wrong thing and didn't find #23, sorry. Thanks again for your efforts.

**Comment by ralight** [comment]
No problem, it's useful to know the level of people being affected.
